fil_actor_interface/src/builtin/reward/mod.rs

```diff
@@ -100,4 +106,52 @@ impl State {
             State::V13(st) => from_token_v4_to_v2(st.into_total_storage_power_reward()),
         }
     }
+
+    /// The baseline power the network is targeting at this state's epoch.
+    pub fn this_epoch_baseline_power(self) -> StoragePower {
+        match self {
+            State::V8(st) => st.this_epoch_baseline_power,
+            State::V9(st) => st.this_epoch_baseline_power,
+            State::V10(st) => st.this_epoch_baseline_power,
+            State::V11(st) => st.this_epoch_baseline_power,
+            State::V12(st) => st.this_epoch_baseline_power,
+            State::V13(st) => st.this_epoch_baseline_power,
+        }
+    }
+
+    pub fn deal_provider_collateral_bounds(
```

Comment:
Yeah, it's indeed defined. Just realized that this got moved after `v7` to a new place in Lotus. 

However I don't think we have this defined for anything lower than `v11`. Which is what I'm going to fix.
Also, the code is exactly the same for all the versions. I'm inclined to create a shared helper for this as opposed to copy-pasting the same code like it's done in Lotus.

Reply:
I'd avoid that helper. The vast majority of this repository is copy-pasted and duplicated across versions. Refactoring it may be nice, but it'll make it more difficult to deal with future versions; instead of just copying the code from `builtin-actors`, you'd have to check each function (manually) if it's not already defined elsewhere _and_ if there are zero changes to it.